packit_service/worker/parser.py

```diff
@@ -1133,9 +1133,19 @@ def parse_data_from_testing_farm(
             tuple: project_url, ref, result, summary, copr_build_id,
                 copr_chroot, compose, log_url, identifier
         """
-        result: TestingFarmResult = TestingFarmResult(
-            nested_get(event, "result", "overall") or event.get("state") or "unknown"
-        )
+        tf_state = event.get("state")
+        tf_result = nested_get(event, "result", "overall")
+
+        logger.debug(f"TF payload: state = {tf_state}, result['overall'] = {tf_result}")
+
+        # error and complete are the end states
+        if tf_state not in ("complete", "error"):
+            result = TestingFarmResult(tf_state)
+        else:
+            result = TestingFarmResult(tf_result or tf_state)
+
+        result = result or TestingFarmResult.unknown
```

Comment:
```suggestion
            result = TestingFarmResult(tf_result or tf_state or "unknown")
```

I guess you wanted this

Reply:
I wanted to use it as default even in the `if` branch